src/main/java/it/gov/pagopa/cgn/portal/service/DiscountService.java

```diff
@@ -458,6 +458,12 @@ private void validateTestingDiscount(ProfileEntity profileEntity,
     private void commonDiscountValidation(ProfileEntity profileEntity,
                                           DiscountEntity discountEntity,
                                           boolean isBucketFileChanged) {
+
+        if (discountEntity.getProducts().size()>2){
+            throw new InvalidRequestException(
+                    "Discount cannot have more than 2 product categories");
+        }
```

Comment:
Can we add a check on DiscountEntity with `@Max` annotation on `products` instead ?

Reply:
I decided not to add a constraint via annotation to avoid possible problems deuring the deploy since on DB there are discuntEntities with more than 2 categories present.
The anntation is already present
     @Size(min = 1)
    private List<DiscountProductEntity> products;
    
it should be just updated to   @Size(min = 1, max =2, message ="Discount cannot have more than 2 product categories")

Reply:
Sounds good. Maybe we can decide to add it on another PR, after DB alignment